### PR TITLE
Add sample meta data

### PR DESCRIPTION
**Pull request by barrystokman**
### Fixed

**Steps to consider while deploying**
- Configuration changes:
- Documentation updates:
- Inform users by email:

### Review:
- [ ] Code approved by
- [ ] Tests executed on stage by:  (Document the test done with screen shots and description.)
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg_lims/scripts/add_sample_meta_data.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                converted_collection_date,
+            )
+            row[udf] = converted_collection_date
+        if udf == "Region Code":
+            LOG.info("Converting Region Code from int to string format")
+            row[udf] = str(row[udf]).zfill(2)
+        if not sample.udf.get(udf):
+            LOG.info("Creating udf %s on sample %s: %s", udf, sample.id, row[udf])
+        if sample.udf.get(udf) and sample.udf[udf] != row[udf]:
+            if "Lab" in udf:
```
Comment (henningonsbring) [inline cg_lims/scripts/add_sample_meta_data.py:50]: Can you put a list with constants in the beginning of the code like:
ONLY_ADD_IF_DATA_IS_MISSING = ["Lab Code", "Original Lab", "Original Lab Address"]

then something like:
if udf in ONLY_ADD_IF_DATA_IS_MISSING:
    #What you before did if "Lab" in udf

File: cg_lims/scripts/add_sample_meta_data.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                LOG.warning(
+                    "Lab udf %s on sample (%s) differs from meta data (%s), sample will NOT be "
```
Comment (henningonsbring) [inline cg_lims/scripts/add_sample_meta_data.py:52]: if following my advice regarding list with constants then maybe remove the word "Lab" from the beginning of this string and only say "udf %s on sample..."

Just saying this to make the code more readable, now it is rather confusing to see 'if "Lab" in udf:'

**Review by henningonsbring — APPROVED** [review]

File: cg_lims/scripts/add_sample_meta_data.py (hunk 1 above)
Comment (barrystokman) [reply cg_lims/scripts/add_sample_meta_data.py]: fixed!

File: cg_lims/scripts/add_sample_meta_data.py (hunk 2 above)
Comment (barrystokman) [reply cg_lims/scripts/add_sample_meta_data.py]: fixed

**Comment by barrystokman** [comment]
ran the script on stage:
`lims -c ~/.genologics.yaml scripts -l meta_data.log set-sample-meta-data --data-file /home/glsai/backlogg_sarscov2.csv`
[Uploading meta_data.log…]()

image

**Comment by barrystokman** [comment]
@mayabrandi i removed the script from the CLI, and it ran successfully on Friday. Shall we merge and close this?

**Review by mayabrandi — APPROVED** [review]

**Comment by mayabrandi** [comment]
@barrystokman yes. go ahed and merge

**Pull request merged**